src/streaming_decoder.rs

```diff
@@ -39,10 +39,35 @@ impl<READ: Read> StreamingDecoder<READ, FrameDecoder> {
         decoder.init(&mut source)?;
         Ok(StreamingDecoder { decoder, source })
     }
+}
 
-    pub fn inner(self) -> FrameDecoder {
+impl<READ: Read, DEC: BorrowMut<FrameDecoder>> StreamingDecoder<READ, DEC> {
+    /// Destructures this object into the inner [FrameDecoder].
+    pub fn inner(self) -> DEC {
         self.decoder
     }
+
+    /// Gets a reference to the underlying reader.
+    /// 
+    /// It is inadvisable to directly read from the underlying reader.
```

Comment:
I think the advise is only necessary on the `get_mut`, since reading requires a mutable reference to the reader.